src/mlpack/methods/neighbor_search/neighbor_search.hpp

```diff
@@ -150,10 +153,70 @@ class NeighborSearch
    * @param epsilon Relative approximate error (non-negative).
    * @param metric Instantiated distance metric.
    */
-  NeighborSearch(Tree* referenceTree,
-                 const NeighborSearchMode mode = DUAL_TREE_MODE,
-                 const double epsilon = 0,
-                 const MetricType metric = MetricType());
+  mlpack_deprecated NeighborSearch(
+      Tree* referenceTree,
+      const NeighborSearchMode mode = DUAL_TREE_MODE,
+      const double epsilon = 0,
+      const MetricType metric = MetricType());
```

Comment:
Yes, I agree with you. Done in https://github.com/mlpack/mlpack/pull/765/commits/db16a10803e3107a2fd96ae0d196f8f8685cf076